lib/flake-utils.nix

```diff
@@ -0,0 +1,63 @@
+{ nixpkgs, systems }:
+rec {
+  /* eachDefaultSystem using defaultSystems
+     Usage in a nix flake:
+       {
+         description = "Flake demo";
+
+         inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+
+         outputs = { self, nixpkgs, flake-utils }:
+           nixpkgs.lib.eachDefaultSystem (pkgs:
+             rec {
+               packages = {
+                 hello = pkgs.hello;
+               };
+               defaultPackage = packages.hello;
+             }
+           );
+       }
+
+     Type: eachDefaultSystem :: (attrset -> attrset) -> attrset
+
+     Example:
+       nixpkgs.lib.eachDefaultSystem (pkgs: {
+         defaultPackage = packages.hello;
+       })
+       => {
+         defaultPackage = {
+           aarch64-linux.hello = <<derivation /nix/store/...-hello-2.10.drv>>,
+           ...
+           x86_64-darwin.hello = <<derivation /nix/store/...-hello-2.10.drv>>,
+           x86_64-linux.hello = <<derivation /nix/store/...-hello-2.10.drv>>,
+         };
+       }
+  */
+  eachDefaultSystem = eachSystem systems;
+
+  /* Builds a map from <attr>=value to <attr>.<system>=value for each system.
+     See `eachDefaultSystem` for a full example
+
+     Type: eachSystem :: [string] -> (attrset -> attrset)
+
+     Example:
+       eachSystem ["x86_64-linux"] (pkgs: { hello = 42; })
+       => { hello = { x86_64-linux = 42; }; }
+  */
+  eachSystem = systems: f:
```

Comment:
unrelated but flake-utils has a defaultSystems / allSystems we can pass on to eachSystem: does nixpkgs have an equivalent ?

Reply:
You can get it from `lib.systems.examples` but also orthogonal of what this pr is doing. You can still add those lists later and feed them to `eachSystem`.